File: examples/python/basic.py
"""
Card Deck Shuffler API - Basic Usage Example

This example demonstrates the basic usage of the Card Deck Shuffler API.
API Documentation: https://docs.apiverve.com/ref/carddeckshuffler
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/carddeckshuffler'

def call_carddeckshuffler_api():
    """
    Make a GET request to the Card Deck Shuffler API
    """
    try:
        # Query parameters
        params &#x3D; {&#x27;decks&#x27;: 1, &#x27;jokers&#x27;: false, &#x27;method&#x27;: &#x27;fisher-yates&#x27;}

        headers = {
            'x-api-key': API_KEY
        }

        response = requests.get(API_URL, headers=headers, params=params)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Card Deck Shuffler API...\n')

    result = call_carddeckshuffler_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: npm/examples/basic.js
/**
 * Basic Example - Card Deck Shuffler API
 *
 * This example demonstrates how to use the Card Deck Shuffler API.
 * Make sure to set your API key in the .env file or replace '[YOUR_API_KEY]' below.
 */

require('dotenv').config();
const carddeckshufflerAPI = require('../index.js');

// Initialize the API client
const api = new carddeckshufflerAPI({
    api_key: process.env.API_KEY || '[YOUR_API_KEY]'
});

// Example query
var query = {
  decks: 1,
  jokers: false,
  method: "fisher-yates"
};

// Make the API request using callback
console.log('Making request to Card Deck Shuffler API...\n');

api.execute(query, function (error, data) {
    if (error) {
        console.error('Error occurred:');
        if (error.error) {
            console.error('Message:', error.error);
            console.error('Status:', error.status);
        } else {
            console.error(JSON.stringify(error, null, 2));
        }
        return;
    }

    console.log('Response:');
    console.log(JSON.stringify(data, null, 2));
});
